### PR TITLE
Move away from using **kwargs in python Kafka API.

### PR DESCRIPTION
**Pull request by jcferretti**
Thanks to Jianfeng for pointing out.

### CODE REVIEW

File: Generators/src/main/java/io/deephaven/pythonPreambles/KafkaToolsPreamble.txt [hunk 1]
```diff
@@ -73,56 +74,58 @@ def _passThrough(wrapped, instance, args, kwargs):
...
-        fieldNamesArray = jpy.array('java.lang.String', dict.keys())
-        columnNamesArray = jpy.array('java.lang.String', dict.values())
-        mapping = _java_type_.fieldNameMappingFromParallelArrays(fieldNamesArray, columnNamesArray)
-        return _java_type_.avroSchemaToColumnDefinitions(schema, mapping)
-    raise Exception('too many arguments: ' + len(args))
+def _custom_avroSchemaToColumnDefinitions(schema, mapping:dict = None):
+    if mapping is None:
+        return _java_type_.avroSchemaToColumnDefinitions(schema)
+
+    if not isinstance(mapping, dict):
```
Comment (cpwright) [inline Generators/src/main/java/io/deephaven/pythonPreambles/KafkaToolsPreamble.txt:81]: Still necessary with that mapping:dict specifier?

File: Generators/src/main/java/io/deephaven/pythonPreambles/KafkaToolsPreamble.txt [hunk 2]
```diff
@@ -131,68 +134,69 @@ def consumeToTable(*args, **kwargs):
...
     elif not isinstance(offsets, jpy.JType):
         raise Exception(
-            "type " + type(offsets).__name__ +
-            "  of keyword argument 'offsets' not recognized; only str or dict allowed")
+            "value " + str(offsets) + " of type " + type(offsets).__name__ +
+            "  not recognized for argument 'offsets'; only str, dict or predefined constants allowed")
 
-    key = kwargs.pop('key', FROM_PROPERTIES)
-    value = kwargs.pop('value', FROM_PROPERTIES)
+    if key is None:
```
Comment (cpwright) [inline Generators/src/main/java/io/deephaven/pythonPreambles/KafkaToolsPreamble.txt:140]: It seems like you should use FROM_PROPERTIES in the method signature default.

File: Generators/src/main/java/io/deephaven/pythonPreambles/KafkaToolsPreamble.txt [hunk 3]
```diff
@@ -206,43 +210,37 @@ def avro(schema, **kwargs):
...
 
 @_passThrough
-def simple(*args):
-    if len(args) < 1 or len(args) > 2:
-        raise Exception("one or two arguments expected, instead got " + len(args))
-    colName = args[0]
-    if not _isStr(colName):
-        raise Exception("column_name argument needs to be of str type, instead got " + colName)
+def simple(column_name:str, data_type_str:str = 'string'):
+    if not _isStr(column_name):
```
Comment (cpwright) [inline Generators/src/main/java/io/deephaven/pythonPreambles/KafkaToolsPreamble.txt:233]: still needed?

File: Generators/src/main/java/io/deephaven/pythonPreambles/KafkaToolsPreamble.txt (hunk 2 above)
Comment (jcferretti) [reply Generators/src/main/java/io/deephaven/pythonPreambles/KafkaToolsPreamble.txt:140]: We should, but is sad and we can't.  FROM_PROPERTIES is an object that is not initialized by the time the def definition is processed.   If we say `def(..., some_arg=SOME_CONSTANT, ...)` for a `SOME_CONSTANT` that we initialize from a Java object static via jpy, when python processed the `def` it finds the initial value of `None` for the constant and default arguments are never evaluated again, only once at the time the def is processed, so it doesn't work.   See also https://illumon.slack.com/archives/C028TLCCSPR/p1629244827144200

File: Generators/src/main/java/io/deephaven/pythonPreambles/KafkaToolsPreamble.txt (hunk 1 above)
Comment (jcferretti) [reply Generators/src/main/java/io/deephaven/pythonPreambles/KafkaToolsPreamble.txt:81]: Yes, there is no enforcement from python, only warnings from linters/IDEs.

**Pull request merged**